### PR TITLE
Service endpoint -> Deployments

### PR DESCRIPTION
**Pull request by slinkydeveloper**
See https://github.com/restatedev/restate/issues/989

### CODE REVIEW

File: docs/quickstart.mdx [hunk 1]
```diff
@@ -107,12 +107,12 @@ docker run --name restate_dev --rm -d --network=host docker.io/restatedev/restat
...
-and providing it with the endpoint of the service `http://localhost:8080`.
+We do this by sending a request to the endpoint of the runtime at `http://localhost:8081/deployments`
+and providing it with the deployment endpoint of the service `http://localhost:8080`.
 ```shell
-curl -X POST http://localhost:9070/endpoints -H 'content-type: application/json' -d '{"uri": "http://localhost:9080"}'
+curl -X POST http://localhost:9070/deployments -H 'content-type: application/json' -d '{"uri": "http://localhost:9080"}'
 ```
-Restate then sends a request to the service endpoint to ask which services and methods
+Restate then sends a request to the deployment to ask which services and methods
 are running behind this endpoint and will keep track of these (including of the method signatures).
```
Comment (tillrohrmann) [inline docs/quickstart.mdx:116]:
```suggestion
are running behind this deployment and will keep track of these (including of the method signatures).
```

**Review by tillrohrmann — COMMENTED** [review]
Thanks for creating this PR @slinkydeveloper. The changes look good to me. One question that came up was whether we should rather use "service deployment" instead of "deployment" alone in the documentation texts. Maybe this makes it clearer that we are talking about service deployment and not a server deployment?

**Comment by slinkydeveloper** [comment]
> service deployment

I think you're right on this.

**Comment by slinkydeveloper** [comment]
uhhhhh if we change to `service_deployment`, then how about the admin REST resource? should that be `ServiceDeployment` (or `ServicesDeployment`) as well?

**Comment by slinkydeveloper** [comment]
I did another pass at this, sometimes using the term Service deployment

File: docs/services/upgrades-removal.md [hunk 2]
```diff
@@ -94,24 +94,24 @@ When updating Keyed and Singleton services, the new revisions will continue to u
...
+6. Remove the deployment containing the service with:
 
 ```bash
-$ curl -X DELETE <RESTATE_META_ENDPOINT>/endpoints/{ENDPOINT_ID}?force=true
+$ curl -X DELETE <RESTATE_META_ENDPOINT>/deployments/{DEPLOYMENT_ID}?force=true
 ```
 
 :::info
-Currently Restate doesn't implement all the checks to safely remove a service endpoint, so you have to force the removal through the `force` query parameter.
+Currently Restate doesn't implement all the checks to safely remove a deployment, so you have to force the removal through the `force` query parameter.
```
Comment (tillrohrmann) [inline docs/services/upgrades-removal.md:116]:
```suggestion
Currently Restate doesn't implement all the checks to safely remove a service deployment, so you have to force the removal through the `force` query parameter.
```

File: docs/tour.mdx [hunk 3]
```diff
@@ -149,29 +149,29 @@ When you are finished with the tour, you can stop Restate with `docker stop rest
 
 ### Registering services with Restate
 
-When Restate first comes up, it doesn't know anything about our services. Let's tell it where they are running. You can trigger service discovery by calling the Restate admin API informing it about the endpoint where our service is listening.
+When Restate first comes up, it doesn't know anything about our services. Let's tell it where they are running. You can trigger service discovery by calling the Restate admin API informing it about the deployment endpoint where our service is listening.
```
Comment (tillrohrmann) [inline docs/tour.mdx:152]:
```suggestion
When Restate first comes up, it doesn't know anything about our services. Let's tell it where they are running. You can trigger service discovery by calling the Restate admin API informing it about the service deployment where our service is listening.
```

File: docs/tour.mdx [hunk 4]
```diff
@@ -149,29 +149,29 @@ When you are finished with the tour, you can stop Restate with `docker stop rest
...
+curl -X POST http://localhost:9070/deployments -H 'content-type: application/json' -d '{"uri": "http://localhost:9080"}'
 ```
 
 </TabItem>
 <TabItem value="mac" label="macOS">
 
-You do this by sending a request to the Restate endpoint at `http://localhost:9070/endpoints`
-and providing it with the endpoint of the service `http://host.docker.internal:9080`.
+You do this by sending a request to the Restate endpoint at `http://localhost:9070/deployments`
+and providing it with the deployment endpoint of the service `http://host.docker.internal:9080`.
```
Comment (tillrohrmann) [inline docs/tour.mdx:165]: Maybe?

```suggestion
and providing it with the service deployment URL `http://host.docker.internal:9080`.
```

**Review by tillrohrmann — APPROVED** [review]
LGTM. Thanks for updating the PR @slinkydeveloper. +1 for merging.

**Pull request merged**